mesonbuild/backend/ninjabackend.py

```diff
@@ -2973,8 +3028,14 @@ def generate_single_compile(self, target: build.BuildTarget, src,
             i = os.path.join(self.get_target_private_dir(target), compiler.get_pch_name(pchlist[0]))
             arr.append(i)
             pch_dep = arr
-
-        compiler_name = self.compiler_to_rule_name(compiler)
+        # If TASKING compiler family is used and MIL linking is enabled for the target,
+        # then compilation rule name is a special one to output MIL files
+        # instead of object files for .c files
+        key = OptionKey('b_tasking_mil_link')
+        if key in compiler.base_options and target.get_option(key) and src.endswith('.c'):
```

Comment:
I just noticed that this also needs to do the same `in compilers.lang_suffixes['c']` check.